src/ios/SentryCordova.m

```diff
@@ -8,121 +8,245 @@
-                    [self.commandDelegate sendPluginResult:result callbackId:command.callbackId];
-                }];
-            }
+- (void)setReleaseVersionDist:(SentryEvent *)event {
+  if (event.extra[@"__sentry_version"]) {
+    NSDictionary *infoDict = [[NSBundle mainBundle] infoDictionary];
+    event.releaseName =
+        [NSString stringWithFormat:@"%@-%@", infoDict[@"CFBundleIdentifier"],
+                                   event.extra[@"__sentry_version"]];
+  }
+  if (event.extra[@"__sentry_release"]) {
+    event.releaseName =
+        [NSString stringWithFormat:@"%@", event.extra[@"__sentry_release"]];
+  }
+  if (event.extra[@"__sentry_dist"]) {
+    event.dist =
+        [NSString stringWithFormat:@"%@", event.extra[@"__sentry_dist"]];
+  }
+  event.sdk = @{
+    @"name" : SentryCordovaSdkName,
+    @"version" : SentryCordovaVersionString,
+    @"integrations" : @[ @"sentry-cocoa" ]
+  };
+}
+
+- (void)captureEnvelope:(CDVInvokedUrlCommand *)command {
+  [self.commandDelegate runInBackground:^{
+    NSDictionary *envelopeDict = [command.arguments objectAtIndex:0];
+    CDVPluginResult *result =
+        [CDVPluginResult resultWithStatus:CDVCommandStatus_OK
+                            messageAsBool:YES];
+
+    if ([NSJSONSerialization isValidJSONObject:envelopeDict]) {
+      SentrySdkInfo *sdkInfo =
+          [[SentrySdkInfo alloc] initWithDict:envelopeDict[@"header"]];
+      SentryId *eventId = [[SentryId alloc]
+          initWithUUIDString:envelopeDict[@"header"][@"event_id"]];
+      SentryEnvelopeHeader *envelopeHeader =
+          [[SentryEnvelopeHeader alloc] initWithId:eventId andSdkInfo:sdkInfo];
+
+      NSError *error;
+      NSData *envelopeItemData =
+          [NSJSONSerialization dataWithJSONObject:envelopeDict[@"payload"]
+                                          options:0
+                                            error:&error];
+      if (nil != error) {
+        result = [CDVPluginResult resultWithStatus:CDVCommandStatus_ERROR
+                                     messageAsBool:NO];
+      } else {
+        NSString *itemType = envelopeDict[@"payload"][@"type"];
+        if (itemType == nil) {
+          // Default to event type.
+          itemType = @"event";
         }
-    }];
+
+        SentryEnvelopeItemHeader *envelopeItemHeader =
+            [[SentryEnvelopeItemHeader alloc]
+                initWithType:itemType
+                      length:envelopeItemData.length];
+        SentryEnvelopeItem *envelopeItem =
+            [[SentryEnvelopeItem alloc] initWithHeader:envelopeItemHeader
+                                                  data:envelopeItemData];
+
+        SentryEnvelope *envelope =
+            [[SentryEnvelope alloc] initWithHeader:envelopeHeader
+                                        singleItem:envelopeItem];
+
+#if DEBUG
+        [[SentrySDK currentHub] captureEnvelope:envelope];
+#else
+                if ([envelopeDict[@"payload"][@"level"] isEqualToString:@"fatal"]) {
```

Comment:
Do we need this code when deciding whether or not to use:

`  [self.commandDelegate runInBackground:^{` ?